src/backend/distributed/connection/connection_management.c

```diff
@@ -429,11 +429,13 @@ FindAvailableConnection(dlist_head *connections, uint32 flags)
 			continue;
 		}
 
-		if (connection->forceCloseAtTransactionEnd)
+		if (connection->forceCloseAtTransactionEnd &&
+			!connection->remoteTransaction.beginSent)
 		{
 			/*
 			 * This is a connection that should be closed, probabably because
-			 * of old connection options. So we ignore it. It will
+			 * of old connection options. If there was a BEGIN sent on this connection we would
```

Comment:
/*
			 * This is a connection that should be closed, probably because
			 * of old connection options or removing a node. This will
			 * automatically be closed at the end of the transaction. But, if we are still
			 * inside a transaction, we should keep using this connection as long as a remote 
			 * transaction is in progress over the connection. The main use for this case
			 * is having some commands inside a transaction block after removing nodes. And, we
			 * currently allow very limited operations after removing a node inside a 
			 * transaction block (e.g., no placement access can happen).
			 */